modules/gcp/dc/new_domain_admin_user.ps1.tpl

```diff
@@ -43,45 +43,45 @@ function Decrypt-Credentials {
     $headers.Add("Authorization", "Bearer $($token)")
 
     try {
+        "--> Decrypting account_password..."
         $resource = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
         $resource.Add("ciphertext", "${account_password}")
         $response = Invoke-RestMethod -Method "Post" -Headers $headers -Uri $DECRYPT_URI -Body $resource
         $credsB64 = $response."plaintext"
         $DATA."account_password" = [System.Text.Encoding]::ASCII.GetString([System.Convert]::FromBase64String($credsB64))
     }
     catch {
-        "Error decrypting credentials: $_"
+        "--> ERROR: Failed to decrypt credentials: $_"
         return $false
     }
 }
 
 Start-Transcript -path $LOG_FILE -append
 
 if ([string]::IsNullOrWhiteSpace("${kms_cryptokey_id}")) {
-    "Not using encryption"
+    "--> Script is not using encryption for secrets."
 } else {
-    "Using ecnryption key ${kms_cryptokey_id}"
+    "--> Script is using encryption key ${kms_cryptokey_id} for secrets."
     Decrypt-Credentials
 }
 
-Write-Output "================================================================"
-Write-Output "Creating new AD Domain Admin account ${account_name}..."
-Write-Output "================================================================"
-
+"================================================================"
+"Creating new AD Domain Admin account ${account_name}..."
+"================================================================"
 do {
     Try {
         $Retry = $false
         New-AdUser -Name "${account_name}" -AccountPassword (ConvertTo-SecureString $DATA."account_password" -AsPlainText -Force) -Enabled $True -PasswordNeverExpires $True
     }
     Catch [Microsoft.ActiveDirectory.Management.ADServerDownException] {
-        $_.Exception.Message
+        "--> $($_.Exception.Message)"
 
         if ($Elapsed -ge $Timeout) {
-            Write-Output "Error: Timed out trying to create new AD acccount."
-            exit
+            Write-Output "ERROR: Timed out trying to create new AD acccount, exiting..."
+            exit 1

```

Comment:
how does exit behave in powershell? I think it took the last command's return code?  does using Write-Output mask the exit code? do we even need the exit line?